cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/ConfigurationGroupPage/GroupPanel/WidgetCollection/WidgetPanel/WidgetInfoInput/index.tsx

```diff
@@ -70,12 +67,14 @@ const WidgetInfoInputView: React.FC<IWidgetInfoInputProps> = ({ classes, widgetI
       setWidgetInfo(widgetInfo.setIn([widgetID, 'widgetType'], widgetType));
 
       // Change widget-attributes so that it corresponds to the widget-type
+      const comp = WIDGET_FACTORY[widgetType];
+      const widgetAttributes = comp.getWidgetAttributes();
       setWidgetToAttributes(
         widgetToAttributes.set(
           widgetID,
           Map(
-            Object.keys(WIDGET_TYPE_TO_ATTRIBUTES[widgetType]).reduce((acc, curr) => {
-              acc[curr] = '';
+            Object.keys(widgetAttributes).reduce((acc, curr) => {
+              acc[curr] = undefined;
```

Comment:
Why did we change this from empty string to undefined? Having it be set to undefined seems potentially ambiguous if we want to check for existence of the key in acc.

Reply:
If we set the default value to be an empty string, this will cause errors in certain widgets. For instance, some widget attributes such as `options` should not be set to an empty string, since it's a list. Previously, It has caused errors because this lets the widgets run `forEach` on empty string.